server/src/main/java/org/candlepin/audit/ListenerWrapper.java

```diff
@@ -14,60 +14,75 @@
  */
 public class ListenerWrapper implements MessageHandler {
 
     private EventListener listener;
     private static Logger log = LoggerFactory.getLogger(ListenerWrapper.class);
     private ObjectMapper mapper;
-    public ListenerWrapper(EventListener listener, ObjectMapper mapper) {
+    private ClientSession session;
+
+    public ListenerWrapper(EventListener listener, ObjectMapper mapper, ClientSession session) {
         this.listener = listener;
         this.mapper = mapper;
+        this.session = session;
     }
 
     @Override
     public void onMessage(ClientMessage msg) {
-        String body = msg.getBodyBuffer().readString();
-        log.debug("Got event: {}", body);
-
-        // Exceptions thrown here will cause the event to remain in ActiveMQ:
+        String body = "";
         try {
+            // Acknowledge the message so that the server knows that it was received.
+            // By doing this, the server can update the delivery counts which plays
+            // part in calculating redelivery delays.
+            msg.acknowledge();
+            log.debug("ActiveMQ message {} acknowledged for listener: {}", msg.getMessageID(), listener);
+
+            // Process the message via our EventListener framework.
+            body = msg.getBodyBuffer().readString();
+            log.debug("Got event: {}", body);
             Event event = mapper.readValue(body, Event.class);
             listener.onEvent(event);
-        }
-        catch (JsonMappingException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
-        catch (JsonParseException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
-        catch (IOException e) {
-            log.error("Unable to deserialize event object from msg: " + body, e);
-            throw new RuntimeException("Error deserializing event", e);
-        }
 
-        try {
-            msg.acknowledge();
-            log.debug("ActiveMQ message acknowledged for listener: " + listener);
+            // Finally commit the session so that the message is taken out of the queue.
+            session.commit();
         }
-        catch (ActiveMQException e) {
-            log.error("Unable to acknowledge ActiveMQ msg", e);
+        catch (Exception e) {
+            // Log a warning instead of a full stack trace to reduce log size.
+            String messageId = msg == null ? "" : Long.toString(msg.getMessageID());
+            String reason = e.getCause() == null ? e.getMessage() : e.getCause().getMessage();
+            log.error("Unable to process message {}: {}", messageId, reason);
+
+            // If debugging is enabled log a more in depth message.
+            log.debug("Unable to process message. Rolling back client session.\n" + body, e);
```

Comment:
```
log.debug("Unable to process message. Rolling back client session.\n{}", body, e);
```
will save the JVM from having to do the string concatenation unnecessarily.

Reply:
hrm.... I don't think that the debug method supports that does it (i.e Throwable as the last filter arg)? This was moved from another class, and I thought that's why it was this way.

Reply:
Internally, Logback looks at the last argument of a call to `log.debug(String, Object...)` and if it's a Throwable, it does the [right thing](https://github.com/qos-ch/logback/blob/master/logback-classic/src/main/java/ch/qos/logback/classic/spi/LoggingEvent.java#L141).